### DISCUSSION TITLE
Unban Command

### DISCUSSION THREAD

**Discussion by theParkerMiller** [nextcord/nextcord#903] [Q&A] [answered]
I'm currently writing code for a moderation bot. I've struggled upon the issue of unban commands. Do these commands even exist? Please let me know ASAP.

**Comment by DenverCoder1** [comment, marked as answer]
There is a `guild.unban` method in the documentation. Is there anything you are stuck on?

https://docs.nextcord.dev/en/stable/api.html#nextcord.Guild.unban

**Reply by theParkerMiller** [reply]
Is this what the code should look like?

@bot.slash_command()
async def unban(interaction: nextcord.Interaction, user: nextcord.Member, reason: str):
  if not interaction.user.guild_permissions.administrator:
    await interaction.response.send_message("You are not authorized to run this command.")
  else:
      await interaction.response.send_message(f"Unbanned {user.mention}.")
      if logging is True:
         log_channel = bot.get_channel(logschannel)
         await log_channel.send(f"{user.mention} was unbanned by {interaction.user.mention}")
         await user.unban(reason=reason)

**Reply by DenverCoder1** [reply]
Did you try it or get any issues?

I see a _potential_ problem with it which is that banned users are *not* members of the Guild, so you might have an issue using the member option to find them. If what you have doesn't work, you may want to do something like take the ID of the banned user as a parameter, then unban them using the guild object.
i.e.
```py
await interaction.guild.unban(nextcord.Object(id=user_id), reason=reason)
```

**Reply by theParkerMiller** [reply]
Testing Now.

**Reply by theParkerMiller** [reply]
Is it just going to be that line? Or is it being attached to the @bot.slash_command?

**Reply by theParkerMiller** [reply]
Update: It worked after switching (id=user_id) --> (id=user.id). However, there is one error that shows up when running unban:     await self(interaction, *args, **kwargs)
  File "main.py", line 47, in unban
    await user.unban(reason=reason)
AttributeError: 'User' object has no attribute 'unban'

The above exception was the direct cause of the following exception:

nextcord.errors.ApplicationInvokeError: Command raised an exception: AttributeError: 'User' object has no attribute 'unban'

It will still unban said user. Let me know if there is something that can fix it.

Thanks,
Ghost
